### PR TITLE
Update documentation of JsonApiSerializer

### PR DESCRIPTION
**Pull request by felixkiss**
Related to #205

### CODE REVIEW

**Comment by vinkla** [comment]
Maybe we could add an example transformer class?

**Comment by felixkiss** [comment]
We could do that. But there were no changes to the transformers.

You still have exactly the same format as in the [transformers docs](https://github.com/thephpleague/fractal/blob/gh-pages/transformers.md#including-data).

The only difference is, that you have to specify a resourceKey in the constructor, which always was the case with the `JsonApiSerializer`.

On the other hand, it won't hurt. We can include the [BookTransformer](https://github.com/felixkiss/fractal/blob/json-api/test/Stub/Transformer/JsonApiBookTransformer.php) that is also used in the tests.

File: serializers.md [hunk 1]
```diff
@@ -178,38 +178,66 @@ This is why ArraySerialzier is not recommended, but if you are not using meta da
...
 
-That resource key is used to give it a named namespace:
+The resource key is used to give it a named namespace:
 
 ~~~ php
 // Item
 [
-    'book' => [
-        'foo' => 'bar'
+    'data' => [
```
Comment (Art4) [inline serializers.md]: This must be `'data' => {`

File: serializers.md [hunk 2]
```diff
@@ -178,38 +178,66 @@ This is why ArraySerialzier is not recommended, but if you are not using meta da
...
+$resource = new Item($book, new JsonApiBookTransformer(), 'books');
+$resource = new Collection($books, new JsonApiBookTransformer(), 'books');
 ~~~
 
-That resource key is used to give it a named namespace:
+The resource key is used to give it a named namespace:
 
 ~~~ php
 // Item
 [
```
Comment (Art4) [inline serializers.md]: The top level must be an object `{`

File: serializers.md [hunk 3]
```diff
@@ -178,38 +178,66 @@ This is why ArraySerialzier is not recommended, but if you are not using meta da
...
+            'foo' => 'bar'
+        ],
     ],
 ];
 
 // Collection
 [
-    'books' => [
+    'data' => [
         [
```
Comment (Art4) [inline serializers.md]: This must be `{` too. And so on.

File: serializers.md (hunk 1 above)
Comment (felixkiss) [inline serializers.md]: It is a JSON object. The fractal docs include samples in PHP array notation form. The corresponding JSON would have `{`.

File: serializers.md (hunk 3 above)
Comment (felixkiss) [inline serializers.md]: It is a JSON object. The fractal docs include samples in PHP array notation form. The corresponding JSON would have `{`.

File: serializers.md (hunk 2 above)
Comment (felixkiss) [inline serializers.md]: It is a JSON object. The fractal docs include samples in PHP array notation form. The corresponding JSON would have `{`.

Comment (Art4) [inline serializers.md]: Hmm, I see. But it is a little bit confusing that a JSON API example does not show valid JSON API. :confused:

**Comment by felixkiss** [comment]
@jasonlewis What do you think? Should I convert the example output to JSON or leave it as is?

**Comment by jasonlewis** [comment]
Looks good Felix, keep it consistent. If we want to we can always convert them all to a JSON representation. But should be fine as is for now.

**Pull request merged**